src/app/base/components/heuristic-statuses/heuristic-statuses.component.html

```diff
@@ -0,0 +1,28 @@
+<ng-container *ngIf="status !== 'CREATED' && status !== 'FINISHED' && status !== 'FAILED'"
+    [ngTemplateOutlet]="heuristicStatus" 
+    [ngTemplateOutletContext]="{dataTip:'Heuristic was successfully registered.', color: 'gray',statusName:'Initial'}"></ng-container>
+
+<ng-container *ngIf="status === 'CREATED'"
+    [ngTemplateOutlet]="heuristicStatus" 
+    [ngTemplateOutletContext]="{dataTip:'Heuristic is currently being executed.', color: 'yellow',statusName:'Running'}"></ng-container>
+
+<ng-container *ngIf="status === 'FINISHED'"
+    [ngTemplateOutlet]="heuristicStatus" 
+    [ngTemplateOutletContext]="{dataTip:'Heuristic was successfully executed.', color: 'green',statusName:'Finished'}"></ng-container>
+
+<ng-container *ngIf="status === 'FAILED'"
+    [ngTemplateOutlet]="heuristicStatus" 
+    [ngTemplateOutletContext]="{dataTip:'Heuristic ran into errors.', color: 'red',statusName:'Error'}"></ng-container>
```

Comment:
is there a specific reason why we solve this with template outlets instead of assigning the necessary variables in the oninit or onchange functions of the component?

Would reduce the size of the HTML quite a lot & might make it easier to read because of proper highlighting in ts file. Also switch could be utilized there without any issues.

Reply:
Moved the login in a switch case in the .ts file